api/internal/localizer/localizer_test.go

```diff
@@ -299,3 +299,171 @@ patches:
 	lclzr := createLocalizer(t, fSys, "/a/b", "", "/dst")
 	require.Error(t, lclzr.Localize())
 }
+
+func TestLocalizeDirInTarget(t *testing.T) {
+	type testCase struct {
+		name  string
+		files map[string]string
+	}
+	for _, tc := range []testCase{
+		{
+			name: "multi_nested_child",
+			files: map[string]string{
+				"kustomization.yaml": `apiVersion: kustomize.config.k8s.io/v1beta1
+components:
+- delta/epsilon
+kind: Kustomization
+`,
+				"delta/epsilon/kustomization.yaml": `apiVersion: kustomize.config.k8s.io/v1alpha1
+kind: Component
+namespace: kustomize-namespace
+`,
+			},
+		},
+		{
+			name: "recursive",
+			files: map[string]string{
+				"kustomization.yaml": `apiVersion: kustomize.config.k8s.io/v1beta1
+components:
+- delta
+kind: Kustomization
+`,
+				"delta/kustomization.yaml": `apiVersion: kustomize.config.k8s.io/v1alpha1
+components:
+- epsilon
+kind: Component
+`,
+				"delta/epsilon/kustomization.yaml": `apiVersion: kustomize.config.k8s.io/v1alpha1
+kind: Component
+namespace: kustomize-namespace
+`,
+			},
+		},
+		{
+			name: "file_in_dir",
+			files: map[string]string{
+				"kustomization.yaml": `apiVersion: kustomize.config.k8s.io/v1beta1
+components:
+- delta
+kind: Kustomization
+`,
+				"delta/kustomization.yaml": `apiVersion: kustomize.config.k8s.io/v1alpha1
+kind: Component
+patches:
+- path: patch.yaml
+`,
+				"delta/patch.yaml": podConfiguration,
+			},
+		},
+		{
+			name: "multiple_calls",
+			files: map[string]string{
+				"kustomization.yaml": `apiVersion: kustomize.config.k8s.io/v1beta1
+components:
+- delta
+- delta/epsilon
+kind: Kustomization
+`,
+				"delta/kustomization.yaml": `apiVersion: kustomize.config.k8s.io/v1alpha1
+kind: Component
+namespace: kustomize-namespace
+`,
+				"delta/epsilon/kustomization.yaml": `apiVersion: kustomize.config.k8s.io/v1alpha1
+buildMetadata:
+- managedByLabel
+kind: Component
+`,
+			},
+		},
+		{
+			name: "localize_directory_name_when_absent",
```

Comment:
what does "absent" mean in this context? It looks like the name is "localized-files", but without much familiarity with the localize code I'm not sure why that is special.

Reply:
It's referring to an absent remote reference. This test case is testing that a localize directory name - name of directory that we create to store localized files - can be used in the absence of a remote reference. In such a case, `kustomize localize` doesn't need to create a localize directory, and should allow the `target` directory to contain directories with the localize directory name.

I'll rename the test case to make it more comprehensible